packages/wrangler/src/__tests__/parse.test.ts

```diff
@@ -0,0 +1,272 @@
+      json: {
+        name: "complex",
+        spec: {
+          uptime: [1, 2.5, 3],
+          ok: true,
+        },
+      },
+    },
+  ])("$title", ({ text, json }) => {
+    expect(parseJSON(text)).toStrictEqual(json);
+  });
+  test.each([
+    {
+      title: "should fail to parse json with invalid string",
+      json: `\n{\n"version" "1\n}\n`,
+      error: {
+        text: "Unexpected string",
+        location: {
+          file: "config.json",
+          lineText: `"version" "1`,
+          line: 3,
+          column: 9,
+        },
+      },
+    },
+    {
+      title: "should fail to parse json with invalid number",
+      json: `{\n\t"a":{\n\t\t"b":{\n\t\t\t"c":[012345]\n}\n}\n}`,
+      error: {
+        text: "Unexpected number",
+        location: {
+          file: "config.json",
+          lineText: `\t\t\t"c":[012345]`,
+          line: 4,
+          column: 8,
+        },
+      },
+    },
+  ])("$title", async ({ json, error }) => {
+    await expect(async () =>
+      parseJSON(json, "config.json")
+    ).rejects.toMatchObject({
+      detail: error,
+    });
+  });
+});
+
+describe("indexLocation", () => {
+  test.each([
+    {
+      title: "should calculate location from one-line input",
+      input: "",
+      index: 1,
+      location: {
+        line: 1,
+        column: 0,
+        lineText: "",
+      },
+    },
+    {
+      title: "should calculate location from multi-line input",
+      input: `\n{\n\t"hello":"world"\n}\n`,
+      index: 11,
+      location: {
+        line: 3,
+        column: 7,
+        lineText: `\t"hello":"world"`,
+      },
+    },
+    {
+      title: "should calculate location when index is out of bounds",
+      input: "\n\n\n\n",
+      index: 10,
+      location: {
+        line: 5,
+        column: 0,
+        lineText: undefined,
+      },
+    },
+  ])("$title", ({ input, index, location }) => {
+    expect(indexLocation(input, index)).toStrictEqual(location);
+  });
+});
+
+describe("searchLocation", () => {
+  test.each([
+    {
+      title: "should calculate location from one-line match",
+      input: "name = 'coolthing'",
+      search: "coolthing",
+      location: {
+        line: 1,
+        column: 8,
+        length: 9,
+        lineText: "name = 'coolthing'",
+      },
+    },
+    {
+      title: "should calculate location from multi-line match",
+      input: `\n{"versions":[\n\t"1.2.3",\n\t"1.2.4",\n\t"1.2.5"\n]}\n`,
+      search: "1.2.4",
+      location: {
+        line: 4,
+        column: 2,
+        length: 5,
+        lineText: `\t"1.2.4",`,
+      },
+    },
+    {
+      title: "should calculate location from no match",
+      input: "\n{}\n",
+      search: "apple",
+      location: {
+        line: 3,
+        column: 0,
+        length: undefined,
+        lineText: undefined,
+      },
+    },
+  ])("$title", ({ input, search, location }) => {
+    expect(searchLocation(input, search)).toStrictEqual(location);
+  });
+});
```

Comment:
I think we may as well have a test of the `readFile()` too - this will be pretty straightforward if we remove the additional parsing behaviour from it.

Reply:
Note that you would then want to make use of the `runInTempDir()` helper.